src/createContext.lua

```diff
@@ -0,0 +1,92 @@
+local Symbol = require(script.Parent.Symbol)
+local Binding = require(script.Parent.Binding)
+local oneChild = require(script.Parent.oneChild)
+local Children = require(script.Parent.PropMarkers.Children)
+local PureComponent = require(script.Parent.PureComponent)
+
+local function createProvider(context)
+	local Provider = PureComponent:extend("Provider")
```

Comment:
Let's use a regular component here. Because this component is always going to have children, the `shouldUpdate` check provided by `PureComponent` is almost always going to fail.

In general, we should prefer function components or `Component` by default, and fall back to `PureComponent` with profiling if we have performance issues.